src/App.tsx

```diff
@@ -112,45 +245,77 @@ export const App: React.FC = () => {
-              Active
-            </a>
-
-            <a href="#/completed" className="filter__link">
-              Completed
-            </a>
-          </nav>
-
-          {/* don't show this button if there are no completed todos */}
-          <button type="button" className="todoapp__clear-completed">
-            Clear completed
-          </button>
-        </footer>
+        {
+          !!remainingTodos && (
+            <footer className="todoapp__footer">
+              <span className="todo-count">
+                {remainingTodos}
+                {' '}
+                items left
+              </span>
+
+              {/* Active filter should have a 'selected' class */}
+              <nav className="filter">
+                <a
+                  href="#/"
+                  className={classNames(
+                    'filter__link', { selected: filter === Filter.All },
+                  )}
+                  onClick={(event) => handleFilterChange(Filter.All, event)}
+                >
+                  All
+                </a>
+
+                <a
+                  href="#/active"
+                  className={classNames(
+                    'filter__link', { selected: filter === 'active' },
+                  )}
+                  onClick={(event) => handleFilterChange(Filter.Active, event)}
+                >
+                  Active
+                </a>
+
+                <a
+                  href="#/completed"
+                  className={classNames(
+                    'filter__link', { selected: filter === Filter.Completed },
+                  )}
+                  onClick={
+                    (event) => handleFilterChange(Filter.Completed, event)
+                  }
+                >
+                  Completed
+                </a>
+              </nav>
+
+              {/* don't show this button if there are no completed todos */}
+              {remainingTodos && (
+                <button type="button" className="todoapp__clear-completed">
+                  Clear completed
+                </button>
+              )}
+            </footer>
+          )
+        }
       </div>
 
       {/* Notification is shown in case of any error */}
       {/* Add the 'hidden' class to hide the message smoothly */}
-      <div className="notification is-danger is-light has-text-weight-normal">
-        <button type="button" className="delete" />
-
-        {/* show only one message at a time */}
-        Unable to add a todo
-        <br />
-        Unable to delete a todo
-        <br />
-        Unable to update a todo
-      </div>
+      {
+        errorMessage && (
+          <div
+            className="notification is-danger is-light has-text-weight-normal"
+          >
+            {errorMessage}
+            <button
+              type="button"
+              className="delete"
+              onClick={() => handleCloseNotification}
+            />
```

Comment:
The 'onClick' event handler is missing the function call. Change 'onClick={() => handleCloseNotification}' to'onClick={handleCloseNotification}' to correctly call the function.